crates/iota-indexer/src/historical_fallback/client.rs

```diff
@@ -108,11 +117,17 @@ impl HttpRestKVClient {
 
         let base_url = Url::parse(&base_url)?;
 
+        let cache = MokaCacheBuilder::new(cache_size)
+            .time_to_idle(CACHE_TIME_TO_IDLE)
```

Comment:
have you considered using LRU approach for removing cache entries only when the cache is full?

At any case, I think it would be good to set some [max_capacity](https://docs.rs/moka/latest/moka/sync/struct.CacheBuilder.html#method.max_capacity) regardless of whether we will use the `time_to_idle` or not.